src/stem.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+var util = require("./util");
+
+
+/**
+ * Stem
+ *
+ * A class that creates and manages the "insert widget here" buttons a.k.a "stems"
+ * that appear beside an empty block level element.
+ */
+
+function Stem(blockElem, onClick) {
+  util.assert(!blockElem.$stem);
+
+  this.blockElem = blockElem;
+  this.blockElem.$stem = this;
+
+  this.stemButton = this.createDom();
+  this.reposition();
+
+  // Handle click event
+  this.stemButton.addEventListener( 'click', function() {
+    onClick();
+  }, false);
+
+}
+
+// Static "idompotent" function to retrive or create a Stem
```

Comment:
use /** comments (see other code for examples).

don't quote "idempotent" - it is actually idempotent.